porcupine/plugins/filetypes.py

```diff
@@ -252,6 +254,32 @@ def parse_filetype_name(name: str) -> FileType:
     )
 
 
+filetypes_var: tkinter.StringVar
+
+
+def _sync_filetypes_menu(event: object = None) -> None:
+    tab = get_tab_manager().select()
+
+    if isinstance(tab, tabs.FileTab):
+        try:
+            filetype_name: str = tab.settings.get("filetype_name", str)
+        except KeyError:
+            filetype_name = ""
+        if filetypes_var:
+            filetypes_var.set(filetype_name)
+    else:
+        if filetypes_var:
```

Comment:
As far as I can tell, these if statements are unnecessary. The variable can be a `StringVar` or not exist at all. If it doesn't exist, this would `NameError`, and that apparently never happens. If it is a `StringVar`, the condition evaluates to true.

Reply:
Yeah I had that check since I couldn't initialize it until setup() of the filetype plugin.
If I initialized it when I declared it (module level variable) then tk wasn't ready and it threw an error.
but in any case I don't think the checks really would have helped since setup should always run before the sync method is called
So I cleaned that method up - let me know what you think